### PR TITLE
allow jdk/maven tool per job

### PR DESCRIPTION
**Pull request by Ginxo**
**Thank you for submitting this pull request**

related PRs:

- https://github.com/kiegroup/kie-jenkins-scripts/pull/1011

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* <b>a pull request</b> please add comment: <b>Jenkins retest</b> (using <i>this</i> e.g. <b>Jenkins retest this</b> optional but no longer required)
 
* <b>a full downstream build</b> please add comment: <b>Jenkins run fdb</b>
  
* <b>a compile downstream build</b> please  add comment: <b>Jenkins run cdb</b>

* <b>a full production downstream build</b> please add comment: <b>Jenkins execute product fdb</b>

* <b>an upstream build</b> please add comment: <b>Jenkins run upstream</b>
</details>

### CODE REVIEW

**Review by mbiarnes — APPROVED** [review]
+1

**Review by mbiarnes — APPROVED** [review]
LGTM

File: Jenkinsfile.buildchain [hunk 1]
```diff
@@ -7,6 +7,8 @@ additionalExcludedArtifacts = "${env.ADDITIONAL_EXCLUDED_ARTIFACTS?.trim() ?: ''
 checkstyleFile = env.CHECKSTYLE_FILE?.trim() ?: null
 findbugsFile = env.FINDBUGS_FILE?.trim() ?: null
 pr_type = env.PR_TYPE?.trim() ?: null
+build_jdk_tool = env.BUILD_JDK_TOOL?.trim() ?: 'kie-jdk8'
+build_maven_tool = env.BUILD_MAVEN_TOOL?.trim() ?: 'kie-maven-3.6.3'
```
Comment (mareknovotny) [inline Jenkinsfile.buildchain:11]: how is this in line of our maven upgrade to 3.8.1 ? fyi https://issues.redhat.com/browse/BXMSPROD-1339

**Comment by mareknovotny** [comment]
There is still some problem with build chain definition version file
`11:25:41  [ERROR]  version 2.1 is not allowed in definition file. Allowed versions: 2.0`

**Comment by Ginxo** [comment]
@mareknovotny it didn't get https://github.com/kiegroup/droolsjbpm-build-bootstrap/commit/2311c2c73b5929dda6166351d001b6f50566b90c#diff-1a498e068a28a1be3f7366558fdafcd711aedf6a0569022f751b1cee58ca40ed
relaunching it.

**Comment by Ginxo** [comment]
Jenkins retest

File: Jenkinsfile.buildchain (hunk 1 above)
Comment (Ginxo) [reply Jenkinsfile.buildchain:11]: upgrading Jenkinsfiles it is something that has to be done as a part of the https://issues.redhat.com/browse/BXMSPROD-1339:

- https://github.com/kiegroup/droolsjbpm-build-bootstrap/blob/3e968a15ca77bb99d9a1538ba2c4d6f55063fd99/Jenkinsfile.nightly#L8
- https://github.com/kiegroup/droolsjbpm-build-bootstrap/blob/3e968a15ca77bb99d9a1538ba2c4d6f55063fd99/Jenkinsfile.revapi#L17
- https://github.com/kiegroup/droolsjbpm-build-bootstrap/blob/3e968a15ca77bb99d9a1538ba2c4d6f55063fd99/Jenkinsfile.buildchain#L70

@akoufoudakis

**Review by yurloc — APPROVED** [review]

**Review by mareknovotny — APPROVED** [review]

**Pull request merged**